app/services/transient_registration_search_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class TransientRegistrationSearchService
+  def initialize(term, in_progress, pending_payment, pending_conviction_check)
+    @term = term
+
+    @in_progress = in_progress
+    @pending_payment = pending_payment
+    @pending_conviction_check = pending_conviction_check
+  end
+
+  def search(page)
```

Comment:
This may be personal preference, but it might be useful to add a comment explaining that all active conditions must be met for an item be in `criteria` (eg AND, not OR). When I was trying to work this out, I discovered that `merge!` for hashes combines any items that appear in either hash being merged, but `merge!` for ActiveRecord (and I assume also Mongoid) only combines items which appear in _both_ lists. So it's not totally clear from the method name. Might save future people some googling. What do you think?

Reply:
The first point about for an item to be returned in the results, it must match all criteria selected makes sense, along with a note that we allow you to select an invalid combination.

Not sure anything about the merge though is necessary. If I've understood how you say ActiveRecord works then this

```ruby
    criteria = WasteCarriersEngine::TransientRegistration.order_by("metaData.lastModified": :desc)
    criteria.merge!(WasteCarriersEngine::TransientRegistration.search_term(@term)) if @term.present?
```

would fail because the 2 criteria being merged are not the same??? As the code is written and infers that the second criteria is merged with the existing criteria, I'm not sure a comment as well is necessary.

But I might have misunderstood what the issue is.

Reply:
Ahh, I guess I thought of it merging the list of items rather than the list of criteria! Maybe my bad then. But I think if we're agreed about the comment then that's fine.

Reply:
Got it! Now that makes sense. Have pushed a comment, let me know what you think.

Reply:
Awesome, makes way more sense now! (And I learned something about Mongoid.)